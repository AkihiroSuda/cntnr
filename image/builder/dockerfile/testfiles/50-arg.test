FROM alpine:3.7

ARG argd dval
ARG argp
ARG port1 80
ARG port2 443/tcp

ENV MYVARX ${argd}
ENV MYVARY ${argp}

COPY $argd /dest
COPY /src $argd
COPY --chown=$argd /src /dest

EXPOSE $port1 $port2

LABEL $argd=$argp

STOPSIGNAL $argd

USER $argd

VOLUME $argd $argp

WORKDIR $argd

RUN echo $argd

# TODO: test entrypoint/cmd substitution