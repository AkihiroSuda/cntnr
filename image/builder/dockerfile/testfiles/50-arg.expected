FROM alpine:3.7
NAME 0

ENV "MYVARX"="dval"
ENV "MYVARY"="pval"

COPY dir="./ctx" "dval" "/dest" nil
COPY dir="./ctx" "/src" "dval" nil
COPY dir="./ctx" "/src" "/dest" dval
COPY image="dval" "/src" "/dest" nil

EXPOSE 80 443/tcp

LABEL "dval"="pval"

STOPSIGNAL dval

USER dval

VOLUME "dval" "pval"

WORKDIR dval

RUN "argd"="dval" "argp"="pval" "port1"="80" "port2"="443/tcp""/bin/sh" "-c" "echo $argd"

ENTRYPOINT "/bin/sh" "-c" "echo $argd"
ENTRYPOINT "/bin/echo" "$argd"